src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -4595,10 +4595,15 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
 
         val tp = tpt1.tpe
         val sym = tp.typeSymbol.initialize
-        if ((sym.isAbstractType || sym.hasAbstractFlag)
-            && !(sym.isJavaAnnotation && context.inAnnotation))
-          IsAbstractError(tree, sym)
-        else if (isPrimitiveValueClass(sym)) {
+
+        if (tpt.isErroneous) {
+          setError(tpt)
```

Comment:
So this is what prevents the <none> does not extend scala.annotation.Annotation from partest t7259 but it seems to change partest logImplicits. I wasn't sure if this was the right thing to do. Should it not even get to this point? I could also just make this:
```
if (context.typingAnnotationCore && !sym.isJavaAnnotation && !(sym isNonBottomSubClass AnnotationClass) && !tpt.isErroneous) {
          DoesNotExtendAnnotation(tree, sym)
} else if ...
```
but that also doesn't seem proper.